sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -204,6 +204,27 @@ package object util extends Logging {
     truncatedString(seq, "", sep, "", maxFields)
   }
 
+  /** Whether we have warned about plan string truncation yet. */
+  private val planSizeWarningPrinted = new AtomicBoolean(false)
```

Comment:
Such checking can be inside of extension of `StringConcat`. Don't think it is good idea to have the flag here